### ISSUE TITLE
Ctrl+Space in Powershell Core produces double quote character

### ISSUE THREAD

**Issue by jwefers** [microsoft/terminal#6454]
# Environment

```none
Windows build number:  10.0.19041.0
Windows Terminal version (if applicable): 1.0.1402.0 (Preview from Store)
Locale: ENG language, DE keyboard (German)

Any other software?
Powershell Core 7.0.1, 7.1.0-preview.3
```

# Steps to reproduce

Having a german keyboard, open PowerShell Core in Windows Terminal. Start with e.g. `Get-`, then Ctrl+Space.

Also, do the same for normal Windows Powershell 5.1.xxx

# Expected behavior

The autocomplete Menu as it works in normal Powershell terminal shows up.

# Actual behavior

In PS Core: double quote character is typed
In PS: nothing happens, but apparently, the terminal (? console host?) is registering something, as if i repeat it rapidly, the cursor does not blink but stays until i stop
(In WSL, also nothing happens, also registers something)

**Comment by zadjii-msft** [comment]
I'm hoping this will just magically go away with all the work that's been done for #4999 in Terminal 1.1.

**Comment by clericc** [comment]
Ah, i read that thread and what made me open this defect was your remark:
![image](https://user-images.githubusercontent.com/851823/84796513-fe372b00-aff8-11ea-8e1a-732278b77733.png)
that is should already be fixed

**Comment by clericc** [comment]
What i just discover: The very same issue (`"` being typed), happens in the VS Code terminal:
![image](https://user-images.githubusercontent.com/851823/84796940-861d3500-aff9-11ea-8d0c-86e102d87aaa.png)

not your turf probably but coincidence?

**Comment by DHowett** [comment]
So, this is definitely our turf (unfortunately!). We own the translation layer that VSCode is using, and it's the same one as in Windows Terminal.

<kbd>Ctrl+Space</kbd> gets encoded as `NUL`, and on the receiving end we ... translate that to <kbd>Ctrl+Shif+2</kbd>

Powershell treats this the same as <kbd>Shift+2</kbd>.

So, a little more info about terminals.

The don't really communicate in keys and modifiers and all that. They communicate in control characters or character sequences start are introduced with `U+001B ESCAPE`. Unfortunately, a number of different keys map to the same thing.

`NUL` is one of those encodings with multiple keys. <kbd>Ctrl+Space</kbd> and <kbd>Ctrl+Shift+2</kbd> both encode to the same byte. When we try to go back from "byte" to "what keys was the user pressing at the time" we get the wrong one.

We **cannot fix this** in the general case. However, we recently landed #4999 which improves the fidelity of keyboard input for all consumers (opt-in!) by giving them raw access to the literal keys that were pressed.

With 4999, and with Terminal preview 1.1, Ctrl+Space comes out as Ctrl+Space and Ctrl+Shift+2 comes out as Ctrl+Shift+2.

/dup #4999 

:smile:
